src/main.rs

```diff
@@ -84,6 +84,8 @@ fn panic(info: &core::panic::PanicInfo) -> ! {
 extern "C" {
     static _start: u8;
     static _stack_end: u8;
+    static _umode_bin: *const u8;
+    static _umode_bin_len: usize;
```

Comment:
the more correct type for these is `u8` -- it's the address of these symbols that is the start of the u-mode binary and the length.

Reply:
fixed